Source/New/Modules/Slithin.Modules.Sync/SynchronizeImpl.cs

```diff
@@ -0,0 +1,66 @@
+﻿using AuroraModularis.Core;
+using Slithin.Modules.BaseServices.Models;
+using Slithin.Modules.Device.Models;
+using Slithin.Modules.Sync.Models;
+using Slithin.Modules.UI.Models;
+
+namespace Slithin.Modules.Sync;
+
+public class SynchronizeImpl : ISynchronizeService
+{
+    public async void Synchronize()
+    {
+        await Task.Run(() =>
+        {
+            var device = ServiceContainer.Current.Resolve<IRemarkableDevice>();
+            var pathManager = ServiceContainer.Current.Resolve<IPathManager>();
+            var notificationService = ServiceContainer.Current.Resolve<INotificationService>();
+
+            {
+                var status = notificationService.ShowStatus("Synchronizing Device: Fetching Notebooks");
+                var notebooks = device.FetchFilesWithModified("/home/root/.local/share/remarkable/xochitl");
+
+                foreach (var (p, time) in notebooks)
+                {
+                    var path = Path.Combine(pathManager.NotebooksDir, p);
+                    var fileInfo = new FileInfo(path);
+                    if (!fileInfo.Exists || IsFileOlder(fileInfo, time))
+                    {
+                        status.Step($"Sync Notebook: Downloading {p} ...");
+                        device.Download($"/home/root/.local/share/remarkable/xochitl/{p}", fileInfo);
+                    }
+                    else
+                    {
+                        status.Step($"Sync Notebook: Skipping {p} (Up to date)");
+                    }
+                }
+            }
+
+            {
+                var status = notificationService.ShowStatus("Synchronizing Device: Fetching Templates");
```

Comment:
Textausgaben bitte über den ILocalisationService übersetzen und die neuen Strings in strings.json adden